libflux/flux-core/src/semantic/tests.rs

```diff
@@ -90,33 +90,22 @@ fn infer_types(
         .map(|(path, pkg)| (path, parse_map(pkg)))
         .collect();
 
-    // Compute the maximum type variable and init fresher
-    let mut max = imports.max_tvar();
-    let mut f = Fresher::from(max.0 + 1);
-
     // Instantiate package importer using generic objects
     let importer: HashMap<&str, PolyType> = imports
         .into_iter()
-        .map(|(path, types)| (path, build_polytype(types, &mut f).unwrap()))
+        .map(|(path, types)| {
+            (
+                path,
+                build_polytype(types, &mut Fresher::default()).unwrap(),
+            )
+        })
         .collect();
 
-    for (_, t) in &importer {
-        max = if t.max_tvar() > max {
-            t.max_tvar()
-        } else {
-            max
-        };
-    }
-
     // Parse polytype expressions in initial environment.
     let env = parse_map(env);
 
     // Compute the maximum type variable and init fresher
-    let max = if env.max_tvar() > max {
-        env.max_tvar()
-    } else {
-        max
-    };
+    let max = env.max_tvar();
```

Comment:
Is this needed anymore? It seems to me like the area below that initializes a fresher should also use `Fresher::default()`.

Reply:
It shouldn't be, however the `imports` test seems to require that `readwrite = true` right now which makes the substitutions also be applied to that top level environment which shouldn't be necessary as there should not be any free variables in it anyway however it appears to preserve the type as `"f" => "(x: A) => A"` else it gets changed to the isomorphic `"f" => "(x: D) => D"`. Not really sure how that works though.

Reply:
We should see if we can fix that. I think this section of the code is the part that needs to be changed for the embedding semantic graph to work.

Reply:
I believe I have fixed this now, though I did have to add in another "bug" error to guard against variable reuse (i.e a code path I do not believe should ever be taken).

Reply:
Seems to be some issues in the go-based tests still unfortunately